Salim/multi chain token rate and balances

salimtb opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**


https://github.com/user-attachments/assets/88614f56-dade-4359-961c-7cb3df4e8ae5



<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

gambinish committed:
feat: Add sortAssets utility function

gambinish committed:
chore: Breakup monolithic Tokens/index.tsx into smaller components

gambinish committed:
fix: Correctly handle null check for buyable footer button

gambinish committed:
fix: Tweak boolean logic for rendering footer components

gambinish committed:
chore: Update snapshot

gambinish committed:
chore: Type out tokens as TokenI[]

gambinish committed:
chore: Remove unecessary comments around import statements in Networth component

gambinish committed:
Merge branch 'main' into chore/componetize-tokens-screen

gambinish committed:
Merge branch 'main' into chore/componetize-tokens-screen

gambinish committed:
Merge branch 'main' into chore/componetize-tokens-screen

gambinish committed:
chore: Add unit tests to utility file, provide exports from index file

gambinish committed:
chore: Rename handleBalance to me more descriptive of what its logic is handling

gambinish committed:
chore: rename Networth to PortfolioBalance for clarity

gambinish committed:
chore: Cleanup styles extra whitespace

gambinish committed:
Merge branch 'main' into chore/componetize-tokens-screen

gambinish committed:
fix: Update snapshot tests

gambinish committed:
Merge branch 'chore/componetize-tokens-screen' of github.com:MetaMask/metamask-mobile into chore/componetize-tokens-screen

gambinish committed:
fix: Remove unused component

gambinish committed:
Merge branch 'main' into chore/componetize-tokens-screen

gambinish committed:
chore: Move PotfolioBalance above TabView

gambinish committed:
fix: Wrap PortfolioBalance and ScrollableView in fragment for token rendering

gambinish committed:
fix: Padding tweak below tab bar for consistency of token list

gambinish committed:
fix: Reorganize test suite for PortfolioBalance

gambinish committed:
fix: Revert changes to wallet test

gambinish committed:
fix: Update fiatBalance text style to DisplayMD

gambinish committed:
chore: Patch PreferencesController with tokenSortConfig

gambinish committed:
chore: Patch preferences controller with type

gambinish committed:
feat: Build token sorting UI, patch PreferencesController

gambinish committed:
chore: Merge main, address merge conflicts

gambinish committed:
fix: ListHeaderComponent not needed in FlatList anymore

gambinish committed:
fix: Update snapshots and lint

gambinish committed:
chore: Merge develop, address conflicts

gambinish committed:
chore: Remove destructive index

gambinish committed:
Merge branch 'main' into feat/mmassets_357-sort-import-tokens-mobile--reorg-portfolio-balance

gambinish committed:
chore: Merge main, address conflicts

gambinish committed:
chore: Repatch preferences controller with tokenSortConfig

gambinish committed:
fix: Declning balance sort

gambinish committed:
fix: Export raw value from deriveBalanceFromAssetMarketDetails

gambinish committed:
fix: deriveBalanceFromAssetMarketDetails.test.ts

gambinish committed:
fix: Update unit tests and snapshots

gambinish committed:
fix: Lint unused imports from Tokens/index.tsx

gambinish committed:
chore: Update testIds and locale strings

gambinish committed:
fix: Update WalletView import selector

gambinish committed:
fix: Update snapshots

gambinish committed:
chore: Cleanup

gambinish committed:
fix: Add navigation type

salimtb committed:
fix: fix snapshot

salimtb committed:
Merge branch 'main' into feat/mmassets_357-sort-import-tokens-mobile--reorg-portfolio-balance

gambinish committed:
chore: Update snapshots

gambinish committed:
Merge branch 'feat/mmassets_357-sort-import-tokens-mobile--reorg-portfolio-balance' into feat/mmassets_357-sort-import-tokens-mobile--token-sort-config

gambinish committed:
fix: Update placeholders in switch statement

gambinish committed:
Merge branch 'feat/mmassets_357-sort-import-tokens-mobile--token-sort-config' of github.com:MetaMask/metamask-mobile into feat/mmassets_357-sort-import-tokens-mobile--token-sort-config

gambinish committed:
chore: Add additional e2e test for import token footer link

gambinish committed:
chore: Update snapshots

gambinish committed:
chore: Patch PreferencesController with tokenNetworkFilter

gambinish committed:
fix: Update to correct type in PreferenceController patch

gambinish committed:
fix: Correct method name in patched PreferenceController

gambinish committed:
chore: Integrate Filter dropdown with PreferencesController

gambinish committed:
fix: Break out of switch statement rather than returning arbitrary string

gambinish committed:
chore: Init BottomSheet refactor

gambinish committed:
fix: Remove arbitrary 8 second delays from import-tokens.spec

gambinish committed:
fix: Use ButtonBase component to customize button text color more directly

gambinish committed:
fix: Update snapshot

gambinish committed:
fix: Mock initial state that PortfolioBalance component needs

gambinish committed:
fix: Mock tabs in snapshot

gambinish committed:
fix: Consider Error boundary during Wallet rendering process

gambinish committed:
fix: Update localized string

gambinish committed:
Merge branch 'feat/mmassets_357-sort-import-tokens-mobile--reorg-portfolio-balance' into feat/mmassets_357-sort-import-tokens-mobile--token-sort-config

gambinish committed:
Merge branch 'main' into feat/mmassets_357-sort-import-tokens-mobile--reorg-portfolio-balance

gambinish committed:
Merge branch 'feat/mmassets_357-sort-import-tokens-mobile--reorg-portfolio-balance' into feat/mmassets_357-sort-import-tokens-mobile--token-sort-config

gambinish committed:
Merge branch 'main' into feat/mmassets_357-sort-import-tokens-mobile--reorg-portfolio-balance

gambinish committed:
Merge branch 'feat/mmassets_357-sort-import-tokens-mobile--reorg-portfolio-balance' into feat/mmassets_357-sort-import-tokens-mobile--token-sort-config

gambinish committed:
Merge branch 'main' into feat/mmassets_357-sort-import-tokens-mobile--token-sort-config

gambinish committed:
chore: Add TokenSortBottomSheet to RootModalFlow in Routes

gambinish committed:
fix: Resolve upstream conflicts

gambinish committed:
fix: Lint:tsc

gambinish committed:
fix: Lint

gambinish committed:
chore: Init unit tests for new UI

gambinish committed:
chore: Merge main

gambinish committed:
chore: Show selected network in filter

gambinish committed:
fix: Update unit tests and snapshots

gambinish committed:
chore: Merge main, address conflicts

gambinish committed:
fix: Add test for refresh control on tokens screen

gambinish committed:
chore: Update testID selectors for token list

gambinish committed:
chore: Add tokenSort BottomSheet trigger test

gambinish committed:
chore: Lint

gambinish committed:
fix: Update snapshots

gambinish committed:
chore: TokenSortBottomSheet unit tests

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
chore: Cleanup

gambinish committed:
chore: Cleanup, and queryByTestID for test stability

gambinish committed:
chore: More cleanup

gambinish committed:
chore: Merge main, address conflicts

gambinish committed:
fix: Lint

gambinish committed:
Merge branch 'feat/mmassets_357-sort-import-tokens-mobile--token-sort-config--bottom-sheet' into feat/mmassets-431_network-filter-ui

gambinish committed:
fix: Lint after conflicts

gambinish committed:
fix: Missing testIDs after merge for unit tests

gambinish committed:
chore: Merge main, address conflicts

gambinish committed:
refactor: Filter ActionSheet should be BottomSheet

gambinish committed:
chore: Tweak styles to fix merge conflict

gambinish committed:
fix: Update unit tests with correct import path

gambinish committed:
chore: Add unit tests for TokenFilterBottomSheet

gambinish committed:
chore: Add locale strings

gambinish committed:
chore: Add MM_TOKEN_NETWORK_FILTER_ENABLED feature flag

gambinish committed:
fix: Add missing locale string

gambinish committed:
chore: Update snapshots

gambinish committed:
fix: Cleanup localisation strings and code comments

gambinish committed:
fix: Cleanup localisation strings, remove code comments, update feature flag to PORTFOLIO_VIEW

gambinish committed:
chore: Revert changes to component-library that are no longer needed

gambinish committed:
fix: Lint whitespace diff

gambinish committed:
fix: Another whitespace diff fix

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
chore: Merge main

gambinish committed:
fix: Remove outdated patch to preferenceController

gambinish committed:
chore: Repatch package with tokenNetworkFilter

gambinish committed:
fix: Lint

gambinish committed:
chore: Merge main, address conflicts

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
chore: Merge main

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
refactor: Update AssetListControls buttons to icons per design revisions

gambinish committed:
fix: Update button text wrapping with snapshots

gambinish committed:
Merge branch 'feat/mmassets-431_network-filter-ui' of github.com:MetaMask/metamask-mobile into feat/mmassets-431_network-filter-ui

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
fix: Lint

gambinish committed:
Merge branch 'feat/mmassets-431_network-filter-ui' of github.com:MetaMask/metamask-mobile into feat/mmassets-431_network-filter-ui

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
chore: Replace unrelated build flag

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
fix: Remove unused font style

gambinish committed:
fix: Use true for feature flag

gambinish committed:
fix: Unblock podfile during CI

gambinish committed:
chore: Merge main, fix conflicts

salimtb committed:
feat: add token rates controller and token balances and token list controller polling

vinnyhoward committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish commented on app/components/hooks/usePolling.ts:
```diff
@@ -0,0 +1,58 @@
+// A hook that manages multiple polling loops of a polling controller.
+// Callers provide an array of inputs, and the hook manages starting
+// and stopping polling loops for each input.
+const usePolling = <PollingInput>(
```
I wonder if we should call this `useMultiPolling` like we are in [extension](https://github.com/MetaMask/metamask-extension/blob/develop/ui/hooks/useMultiPolling.ts) to keep things consistent

gambinish replied:
I see that there is a conflict with `usePolling` that already got merged, so renaming this file will probably remove that conflict as well

salimtb committed:
fix: add patch for controllers

gambinish committed:
fix: Tweak padding

gambinish committed:
Merge branch 'main' into feat/mmassets-431_network-filter-ui

gambinish committed:
chore: Update snapshot

gambinish committed:
Merge branch 'feat/mmassets-431_network-filter-ui' of github.com:MetaMask/metamask-mobile into feat/mmassets-431_network-filter-ui

sethkfman committed:
feat: 7.34.0 (#11858)

This is the release candidate for version 7.34.0
 
  # Team sign-off checklist
  - [ ] team-accounts
  - [ ] team-assets
  - [x] team-confirmations
  - [x] team-design-system
  - [ ] team-lavamoat
  - [x] team-mobile-platform 
  - [x] team-notifications
  - [x] team-ramp
  - [ ] team-sdk
  - [ ] team-security
  - [x] team-snaps-platform 
  - [x] team-stake
  - [ ] team-tiger
  - [ ] team-wallet-framework
  - [ ] team-wallet-ux


  # Reference
- Testing plan sheet -
https://docs.google.com/spreadsheets/d/1tsoodlAlyvEUpkkcNcbZ4PM9HuC9cEM80RZeoVv5OCQ/edit?gid=404070372#gid=404070372

---------

Signed-off-by: Kai Huang <kai.huang@consensys.net>
Co-authored-by: runway-github[bot] <73448015+runway-github[bot]@users.noreply.github.com>
Co-authored-by: MetaMask Bot <37885440+metamaskbot@users.noreply.github.com>
Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Xiaoming Wang <7315988+dawnseeker8@users.noreply.github.com>
Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>
Co-authored-by: Xiaoming Wang <dawnseeker8@gmail.com>
Co-authored-by: Nicholas Ellul <nicholas.ellul1@gmail.com>
Co-authored-by: legobt <6wbvkn0j@anonaddy.me>
Co-authored-by: OGPoyraz <omergoktugpoyraz@gmail.com>
Co-authored-by: Salim TOUBAL <salim.toubal@outlook.com>
Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>
Co-authored-by: tommasini <tommasini15@gmail.com>
Co-authored-by: Bryan Fullam <bryanfullam@gmail.com>
Co-authored-by: Mpendulo Ndlovu <mpendulo@elefantel.com>
Co-authored-by: Kaihuang72490 <147628638+Kaihuang72490@users.noreply.github.com>
Co-authored-by: Cal-L <cleun007@gmail.com>
Co-authored-by: Cal Leung <cal.leung@consensys.net>
Co-authored-by: sahar-fehri <sahar.fehri@consensys.net>
Co-authored-by: Vince Howard <vincenguyenhoward@gmail.com>
Co-authored-by: Matthew Walsh <matthew.walsh@consensys.net>

salimtb committed:
Merge branch 'feat/mmassets-431_network-filter-ui' into salim/multi-chain-token-rate-and-balances

vinnyhoward committed:
Merge branch 'main' into salim/multi-chain-token-rate-and-balances
